src/NServiceBus.Core/Persistence/InMemory/TimeoutPersister/InMemoryTimeoutPersister.cs

```diff
@@ -105,5 +106,16 @@ public void RemoveTimeoutBy(Guid sagaId)
                 readerWriterLock.ExitWriteLock();
             }
         }
+
+        public TimeoutData Peek(string timeoutId)
+        {
+            return storage.SingleOrDefault(t => t.Id == timeoutId);
```

Comment:
no use a try finally and return the value directly. The lock is there to avoid issues in case a second thread tries to modify the underlying list which is not threadsafe. see https://github.com/Particular/NServiceBus/pull/2968/files#diff-9e880742f0ee754bc7e058d0ab752e7aR115